core/src/main/java/com/scalar/db/config/ConfigUtils.java

```diff
@@ -1,22 +1,49 @@
 package com.scalar.db.config;
 
+import com.google.common.annotations.VisibleForTesting;
 import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableMap;
 import java.util.Properties;
+import javax.annotation.Nullable;
+import org.apache.commons.text.StringSubstitutor;
+import org.apache.commons.text.lookup.StringLookupFactory;
 
+/**
+ * A utility class for retrieving configuration values from {@link Properties}. You can use
+ * placeholders in the values, and they are replaced with environment variables
+ * (${env:&#60;environment variable name&#62;}) or system properties (${sys:&#60;system property
+ * name&#62;}). You can also specify default values in placeholders like ${sys:&#60;system property
+ * name&#62;:-defaultValue}.
+ */
 public final class ConfigUtils {
 
+  /**
+   * A {@link StringSubstitutor} instance with the environment variable string lookup and system
+   * property string lookup.
+   */
+  private static final StringSubstitutor stringSubstitutor =
```

Comment:
Use `StringSubstitutor` in Apache Commons Text with the environment variable string lookup and system property string lookup.

By using it, we replace placeholders with environment variables and system properties.